### ISSUE TITLE
circuit breaker example with kubernetes

### ISSUE THREAD

**Issue by daljeet-hcl** [hashicorp/consul#8261]
Circuit breaker implementation while using Consul Connect Annotations in Kubernetes environment #6250 

https://github.com/hashicorp/consul/issues/6250  quoting this issue, I don't find any consul example for circuit breaker (when consul is running on Kubernetes through side car injected pods)  , Can anyone help here ? 

how to implement circuit breaker, if we are using annotations to inject sidecar proxies in Kubernetes pods?

**Comment by javakarthik** [comment]
Could you help us with some reference to implement Circuit breaker with retry and timeout using consul connect on Kubernetes?

**Comment by idrennanvmware** [comment]
I agree that some more complex examples around this could be beneficial

That said, doesn this https://www.consul.io/docs/connect/proxies/envoy#escape-hatch-overrides send you in the right direction?   I believe you're going to need to write your own Envoy config to leverage those features of Envoy

**Comment by javakarthik** [comment]
Thanks for your reply. This seems to be little complex and could not find any working example on kubernetes. Could you please share if you have done any POC?

**Comment by blake** [comment]
As of Consul 1.10.0, it is possible to configure max connections (HTTP/1.1), max pending requests, and max concurrent request limits using the [UpstreamConfig](https://www.consul.io/docs/connect/config-entries/service-defaults#upstreamconfig) field in the [service defaults](https://www.consul.io/docs/connect/config-entries/service-defaults) configuration entry.

An example configuration for Kubernetes can be found here <https://www.consul.io/docs/connect/config-entries/service-defaults#upstream-configuration>.
